doc/source/about/roadmap.md

```diff
@@ -1,5 +1,84 @@
 
-WIP
+### Fully vectorized geometry engine
+
+GeoPandas uses `shapely` as its geometry engine, based on scalar geometries, requiring a
+loop-based implementation of most GeoPandas methods. That comes at a significant
+performance cost, which is being resolved in shapely 2.0, a new major release resulting
+from a complete rewrite of the internals using the vectorized implementation prototyped
+in the `PyGEOS` project. At this moment, GeoPandas supports all `shapely<2.0`,
+`shapely>=2.0` and `PyGEOS` as possible geometry engines, causing friction in the
+development and an uneven performance on the user side.
+
+GeoPandas 1.0 will require `shapely>=2.0` and deprecate both older shapely and `PyGEOS`
+engines. This change should simplify the code base allowing more manageable maintenance
+and a lower entry barrier for new contributors.
+
+### Feature parity with shapely
+
+Even though GeoPandas uses shapely as the geometry engine, not all its functions are
+exposed at a GeoPandas level. A result is a less convenient API and a need to switch
+between `GeoSeries` objects and lists or arrays of geometries, potentially risking the
+data loss or corruption as the CRS is not included in such operations. In the first
+phase, all element-wise operations (e.g. `segmentize`, or `minimum_bounding_circle`)
+should be exposed as `GeoSeries` methods. The feature parity should be reached in the
+second phase, covering all relevant functions.
+
+### Clarity of the API
+
+The first version of the GeoPandas API is nearly ten years old. The PyData ecosystem has
+significantly changed in the meantime, and some of the early decisions may not be
+future-proof anymore. Ahead of GeoPandas 1.0, the API will undergo a revision, ensuring
+that all the necessary deprecation happens before the major release to provide the
+stability of the API for the coming years.
+
+### Prepared geometries
+
+GeoPandas is using spatial indexing for the operations that may benefit from it. Further
+performance gains can be achieved using prepared geometries. Preparation creates a
+spatial index of individual line segments of geometries, greatly enhancing the speed of
+spatial predicates like `intersects` or `contains`. Given that the preparation has
+become less computationally expensive in `shapely` 2.0, GeoPandas should expose the
+preparation to the user but, more importantly, use smart automatic geometry preparation
+under the hood.
+
+### Pruned dependencies
+
+GeoPandas offers functionality for every step of a typical geospatial workflow, from
+reading of the GIS file formats to geometry operations and handling of Coordinate
+Reference Systems (CRS) and transformation of geometries between them. However, GIS IO
+depends on a relatively heavy C++ library `GDAL` and CRS management on another C++
+library `PROJ`, even though not every application based on GeoPandas is necessarily
+geospatial. GeoPandas 1.0 should eliminate the hard dependency on both `GDAL` and `PROJ`
```

Comment:
> Do you mean that the geo part is the variable here, in that GeoPandas may be used for spatial planar operations that don't necessarily relate to geographic data?

Yes. You may have for example football player trajectories on a plane with 0,0 in one corner. Then you don't need pyproj or pyogrio.

> then this fits with our goal to have pure Python (or at least lighter-weight, easier to install, non-GDAL) I/O libs - which still target geospatial workflows.

I think that those are two complimentary goals. This one is based on a discussion with folks around spatialpandas and their needs allowing them to deprecate the project.

Reply:
Yes, agreed, while the result might be a bit the same (no hard dependency on GDAL), it are two complimentary goals: 1) improving the use of geopandas for non-geo use cases, 2) lighter weight installation for geo use cases. 

I think for GDAL we generally have agreement on this. For PROJ / pyproj, that's still something to discuss? (it will require some code changes in geopandas, but it should certainly be doable to make pyproj also a soft dependency)

Reply:
>  For PROJ / pyproj, that's still something to discuss?

Apart from the implementation perspective, do you have anything against that? I think that there are clear use cases + it may resolve some installation troubles. Implementation-wise, it should not be too complex I believe.

Reply:
No, certainly not in the principle, just wondering a bit whether it is worth it given that pyproj is much smaller compared to GDAL. But it should indeed be quite straightforward to make it optional (move the imports inline), and it is certainly true that you you can perfectly use geopandas without it, so then we should just make that at least possible. 
(testing it might be a bit harder, given the widespread use of geodataframes with crs across the tests)